File: 66 Plus One.cpp
//大數加法的極簡化版
class Solution {
public:
    vector<int> plusOne(vector<int> digits) {
        ++digits[digits.size()-1];
        int i = digits.size()-1;
        while(digits[i] == 10){
            digits[i--] = 0;
            if(i == -1){
                digits.insert(digits.begin(), 1);
                break;
            }
            ++digits[i];
        }
        return digits;
    }
};

File: 66 Plus One.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    digits.splice(0,0,0);
    digits[digits.length-1]++;
    for(var i = digits.length-1;i > 0;i--){
        if(digits[i] == 10){
            digits[i-1]++;
            digits[i] = 0;
        }
        else{
            return digits.slice(1, digits.length);
        }
    }
    return digits;

    //overflow
    //return (parseInt(digits.join(''))+1).toString().split('');
};